server/src/main/java/org/opensearch/cluster/decommission/DecommissionService.java

```
@@ -413,6 +417,27 @@ private static void validateAwarenessAttribute(
         }
     }
 
+    private static void ensureToBeDecommissionedAttributeWeighedAway(ClusterState state, DecommissionAttribute decommissionAttribute) {
+        String msg = null;
+        WeightedRoutingMetadata weightedRoutingMetadata = state.metadata().weightedRoutingMetadata();
+        if (weightedRoutingMetadata != null) {
+            WeightedRouting weightedRouting = weightedRoutingMetadata.getWeightedRouting();
+            if (weightedRouting.attributeName().equals(decommissionAttribute.attributeName())) {
+                Double attributeValueWeight = weightedRouting.weights().get(decommissionAttribute.attributeValue());
+                if (attributeValueWeight == null || attributeValueWeight != 0.0) {
+                    msg = "weight for decommissioned attribute is expected to be [0.0] but found [" + attributeValueWeight + "]";
+                }
+            } else {
+                msg = "no weights are specified to attribute [" + decommissionAttribute.attributeName() + "]";
+            }
+        } else {
+            msg = "no weights are set to the attribute. Please set appropriate weights before triggering decommission action";
+        }
+        if (msg != null) {
+            throw new DecommissioningFailedException(decommissionAttribute, msg);
+        }
+    }
```

I think validation are better put as fail fast
```
if (weightedRoutingMetadata == null) {
    throw new DecommissioningFailedException("no weights are set to the attribute. Please set appropriate weights before triggering decommission action");
}
 if (weightedRouting.attributeName().equals(decommissionAttribute.attributeName()) == false) {
    throw new DecommissioningFailedException("blah...");
}
```

---

Done